File: OOP/documentary.cpp
#include "documentary.h"
#include <cstdlib>

// Read documentary from file.
void documentary::Read(FILE *file) {
    film::Read(file);
    fscanf(file, "%d", &duration);
}

// Write documentary to file.
void documentary::Write(FILE *file) {
    fprintf(file, "%s", "documentary ");
    film::Write(file);
    fprintf(file, "%d%s", duration, " ");
}

void documentary::Random() {
	film::Random();
	duration = rand() % 400;
}

File: Dynamic/Documentary.py
from Film import Film
from Utilities import *


class Documentary(Film):
    def __init__(self):
        super().__init__()
        self.duration = 0

    def read(self, params):
        super().read(params)
        self.duration = int(params[3])

    def to_string(self):
        return 'documentary ' + super().to_string() + ' ' + str(self.duration)

    def random(self):
        super().random()
        self.duration = random_int(400)
